source/math/FloatMatrix.ooc

```diff
@@ -123,13 +128,19 @@ FloatMatrix : cover {
 	}
 	transpose: func -> This {
 		t := this take()
-		result := This new(t dimensions swap())
-		resultElements := result elements
-		thisElements := this elements
-		for (y in 0 .. t height)
-			for (x in 0 .. t width)
-				resultElements[y + x * t height] = thisElements[x + y * t width]
-		this free(Owner Receiver)
+		result: This
+		if (t isVector && this _elements owner == Owner Receiver) {
+			result = this
+			result _dimensions = result _dimensions swap()
+		} else {
+			result = This new(t dimensions swap())
+			resultElements := result elements
+			thisElements := this elements
+			for (y in 0 .. t height)
+				for (x in 0 .. t width)
```

Comment:
`width` and `height` are actually a little more expensive to access now that they're 3 lines long, so in the interest of optimization, consider storing the result of `t height` and `t width` in variables just before the loop.